### DISCUSSION TITLE
Avoid unnecessary execution of functions inside a component that isn't rendered

### DISCUSSION THREAD

**Discussion by TangoPJ** [chakra-ui/chakra-ui#8689] [Q&A] [answered]
Hello! I've encountered a small issue.

I have a modal that is rendered on the page depending on the query parameter `?ga4_integration=open`. If I don't have this parameter, the modal isn't rendered, but at the same time, the hooks inside the modal are still executed!

Can I avoid unnecessary execution of functions inside a component that isn't rendered as shown below?

```js
import { AccountsModal } from 'components/modals/accountsModal'
import { useCallback } from 'react'
import { useSearchParams } from 'react-router-dom'

export const OPEN = 'open'
export const GA4_INTEGRATION = 'ga4_integration'

export const GoogleIntegrationAccounts = () => {
  const [searchParams, setSearchParams] = useSearchParams({ [GA4_INTEGRATION]: '' })

  const id = searchParams.get(GA4_INTEGRATION)
  const isOpen = Boolean(id)

  const isModalOpen = isOpen && id === OPEN

  const onCloseModal = useCallback(() => {
    const nextSearchParams = new URLSearchParams(searchParams)
    nextSearchParams.delete(GA4_INTEGRATION)
    setSearchParams(nextSearchParams)
  }, [searchParams, setSearchParams])

  return isModalOpen && <AccountsModal isOpen={isModalOpen} onClose={onCloseModal} />
}
```

Certainly! Here's the translation:

---

Note that the `return` in the function ensures that the component is returned only if `isModalOpen` is `true`.

**Comment by TangoPJ** [comment]
Can somebody review this code and give a feedback please?

**Comment by segunadebayo** [comment, marked as answer]
React still runs hooks if the component is mounted. Do not render it: `{isOpen && <Modal />}`.
